docs/aws/audit/redshiftmonitoring/rules/not_using_default_port_remediation.mdx

```diff
@@ -1,6 +1,230 @@
+
+This script will help you automate the process of ensuring that your Redshift clusters do not use the default port, thereby enhancing security.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Port" column. If any cluster is using the default port (5439), then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Redshift clusters: Use the following AWS CLI command to list all the Redshift clusters in your account:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,Endpoint.Port]' --output text
+   ```
+
+   This command will return a list of all Redshift clusters along with their ports. 
+
+3. Check the port of each cluster: The default port for Redshift clusters is 5439. If the port returned by the above command for any cluster is 5439, it means that the cluster is using the default port.
+
+4. Automate the process: If you have a large number of clusters, it might be more efficient to automate the process using a Python script. You can use the `boto3` library to interact with AWS services. Here is a simple script that checks if any Redshift cluster is using the default port:
+
+   ```python
+   import boto3
+
+   client = boto3.client('redshift')
+
+   response = client.describe_clusters()
+
+   for cluster in response['Clusters']:
+       if cluster['Endpoint']['Port'] == 5439:
+           print(f"Cluster {cluster['ClusterIdentifier']} is using the default port")
+   ```
+   
+   This script will print the identifiers of all clusters that are using the default port.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>